ichnaea/locate/tests/test_provider.py

```diff
@@ -505,3 +512,146 @@ def test_geoip_country(self):
         self.assertEqual(type(location), Country)
         self.assertEqual(location.country_code, bhutan['country_code'])
         self.assertEqual(location.country_name, bhutan['country_name'])
+
+
+class TestFallbackProvider(ProviderTest):
+
+    TestProvider = FallbackProvider
+
+    def setUp(self):
+        super(TestFallbackProvider, self).setUp()
+
+        self.provider.api_key.allow_fallback = True
+
+        requests_patcher = mock.patch('ichnaea.locate.provider.requests.post')
+        self.mock_requests = requests_patcher.start()
+        self.addCleanup(requests_patcher.stop)
+
+        read_config_patcher = mock.patch('ichnaea.locate.provider.read_config')
+        self.mock_read_config = read_config_patcher.start()
+        self.addCleanup(read_config_patcher.stop)
+
+        self.fallback_url = 'http://test.com'
+        self.mock_read_config().get_map.return_value = {
+            'fallback_url': self.fallback_url,
+        }
+
+    def test_successful_call_returns_location(self):
+        response_location = {
+            'lat': 1.0,
+            'lon': 1.0,
+            'accuracy': 100,
+        }
+        response = mock.Mock()
+        response.content = json.dumps(response_location)
+        self.mock_requests.return_value = response
+
+        cells = [{
+            'cid': cid,
+            'radio': Radio.gsm.name,
+            'mcc': GB_MCC,
+            'mnc': 1,
+            'lac': 1,
+        } for cid in range(1, 3)]
+        wifis = [{'key': '001122334455'}, {'key': '112233445566'}]
```

Comment:
Can you use our factory boy test factories to get these cell/wifi keys? Someone told me to use them :))